mllib/src/main/scala/org/apache/spark/mllib/evaluation/RegressionMetrics.scala

```diff
@@ -39,16 +51,17 @@ class RegressionMetrics @Since("1.2.0") (
    *                                  prediction and observation
    */
   private[mllib] def this(predictionAndObservations: DataFrame) =
-    this(predictionAndObservations.map(r => (r.getDouble(0), r.getDouble(1))))
+    this(predictionAndObservations.map(r => (r.getDouble(0), r.getDouble(1), 1.0)))
 
   /**
    * Use MultivariateOnlineSummarizer to calculate summary statistics of observations and errors.
    */
   private lazy val summary: MultivariateStatisticalSummary = {
-    val summary: MultivariateStatisticalSummary = predictionAndObservations.map {
-      case (prediction, observation) => Vectors.dense(observation, observation - prediction)
+    val summary: MultivariateStatisticalSummary = predictionAndObservationsWithWeight.map {
+      case (prediction, observation, weight) =>
+        (Vectors.dense(observation, observation - prediction), weight)
     }.aggregate(new MultivariateOnlineSummarizer())(
-        (summary, v) => summary.add(v),
+        (summary, sample) => summary.add(sample._1, sample._2),
```

Comment:
We should rename this `summary` variable, it is used 3 times for different objects.

Also, `sample` is a bit too generic here, and using the `_xx` methods are not very intuitive. I suggest you unpack the tuple fully: `{ case (currentSummary, (vec, weight)) => currentSummary.add(vec, weight) }`